pinc/ProjectTransition.inc

```diff
@@ -294,21 +293,16 @@ class ProjectTransition
         if (!is_null($pool) && $this->to_state == $pool->project_checkedout_state) {
             // we want the project event to record who it's been
             // checked out to
-            $project = new Project($projectid);
```

Comment:
there wasn't any good reason a new project object was created here?

Reply:
This isn't necessary because the function is passed a `$project` object. I suspect (but haven't dug into the code) that this was here first and then the function was refactored to take a Project object rather than a `$projectid` and this was just left around.

Reply:
That was my idea as well. Just confirming.

Reply:
@srjfoo, this is the source of the bug you found. So despite the project object existing, reloading it from the DB is required to load the right `checkedoutby` person. I'm digging into *why* the project object doesn't have the right values here.

Reply:
Ha! Found, fixed, and commented why this is needed.

Reply:
Excellent!

Reply:
One last force commit to remove code cleanup in this file that was not destined for this PR.